bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ObservationWeatherHandler.java

```diff
@@ -0,0 +1,193 @@
+                continue;
+            }
+        }
+        updateChannels();
+    }
+
+    @Override
+    protected void updateChannels() {
+        FMIResponse response = this.response;
+        if (response == null) {
+            return;
+        }
+        try {
+            Location location = unwrap(response.getLocations().stream().findFirst(),
+                    "No locations in response -- no data? Aborting");
+            Map<String, String> properties = editProperties();
+            properties.put(PROP_NAME, location.name);
+            properties.put(PROP_LATITUDE, location.latitude.toPlainString());
+            properties.put(PROP_LONGITUDE, location.longitude.toPlainString());
+            updateProperties(properties);
+            // All parameters and locations should share the same timestamps. We use temperature to figure out most
+            // recent timestamp which has non-NaN value
+            int lastValidIndex = unwrap(
+                    response.getData(location, ObservationRequest.PARAM_TEMPERATURE).map(data -> lastValidIndex(data)),
+                    "lastValidIndex not available. Bug?");
+            for (Channel channel : getThing().getChannels()) {
+                ChannelUID channelUID = channel.getUID();
+                if (lastValidIndex < 0) {
+                    updateState(channelUID, UnDefType.UNDEF);
+                } else if (channelUID.getIdWithoutGroup().equals(CHANNEL_TIME)) {
+                    String field = ObservationRequest.PARAM_TEMPERATURE;
+                    Data data = unwrap(response.getData(location, field),
+                            "Field %s not present for location %s in response. Bug?", field, location);
+                    updateEpochSecondStateIfLinked(channelUID, data.timestampsEpochSecs[lastValidIndex]);
+                } else {
+                    String field = getDataField(channelUID);
+                    @Nullable
+                    Unit<?> unit = getUnit(channelUID);
+                    if (field == null) {
+                        logger.error("Channel {} not handled. Bug?", channelUID.getId());
+                        continue;
+                    }
+                    Data data = unwrap(response.getData(location, field),
+                            "Field %s not present for location % in response. Bug?", field, location);
+                    updateStateIfLinked(channelUID, data.values[lastValidIndex], unit);
+                }
+            }
+        } catch (IllegalStateException e) {
+            // IllegalStateException: Unexpected (possibly bug) issue with response
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    String.format("%s: %s", e.getClass().getName(), e.getMessage()));
+        } finally {
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    @SuppressWarnings({ "unused", "null" })
+    private static @Nullable String getDataField(ChannelUID channelUID) {
```

Comment:
See above

Reply:
I don't understand the question. No -- it's not necessary to use static. But I consider it good practice since it is not using instance data.